readme.md

```diff
@@ -1,88 +1,202 @@
+  }
+}
+```
 
-The documented JS API follows [Semantic Versioning](https://semver.org).
-Using undocumented features is at you own risk.
+🍿 [Live demo](https://fingerprintjs.com/demo)
 
-## Contribution
+⏱ [How to upgrade from Open Source to Pro in 30 seconds]()
 
-### Development playground
+📕 [FingerprintJS Pro documentation](https://dev.fingerprintjs.com)
 
-```bash
-# Make sure you have Yarn installed
-yarn install
-yarn playground:start # Add '--port 8765' to change the server port
-```
+## Open-source version reference
 
-Then open http://localhost:8080 in a browser.
-It's reloaded every time the source code is changed.
+### Installation
 
-To build the playground distribution code (e.g. to upload to a static server), run:
+The library is shipped in various formats:
 
-```bash
-yarn playground:build
-```
+- Global variable
+    ```html
+    <script
+      async src="https://cdn.jsdelivr.net/npm/@fingerprintjs/fingerprintjs@3/dist/fp.min.js"
+      onload="/* FingerprintJS... */"
+    ></script>
+    ```
+- UMD
+    ```js
+    require(['https://cdn.jsdelivr.net/npm/@fingerprintjs/fingerprintjs@3/dist/fp.umd.min.js'], (FingerprintJS) => {/* ... */});
+    ```
+- EcmaScript module
+    ```js
+    import FingerprintJS from '@fingerprintjs/fingerprintjs';
+    ```
+- CommonJS
+    ```js
+    const FingerprintJS = require('@fingerprintjs/fingerprintjs');
+    ```
 
-The result will appear at `playground/dist`.
+### API
 
-### How to build
+- `FingerprintJS.load({ delayFallback?: number }): Promise<Agent>`
 
-To build the distribution files of the library, run:
+    Builds an instance of Agent and waits a delay required for a proper operation.
+    `delayFallback` is an optional parameter that sets duration (milliseconds) of the fallback for browsers that don't support [requestIdleCallback](https://developer.mozilla.org/en-US/docs/Web/API/Window/requestIdleCallback);
+    it has a good default value which we don't recommend to change.
+
+- `agent.get({ debug?: boolean }): Promise<{ visitorId: string, components: {/* ... */} }>`
+
+    Gets the visitor identifier.
+    `debug: true` prints debug messages to the console.
+    `visitorId` is the visitor identifier.
+    `components` is a dictionary of components that have formed the identifier;
+    each value is an object like `{ value: any, duration: number }` in case of success
+    and `{ error: object, duration: number }` in case of an unexpected error during getting the component.
+
+- `FingerprintJS.hashComponents(components: object): string`
+
+    Converts a dictionary of components (described above) into a short hash string a.k.a. a visitor identifier.
+    Designed for extending the library with your own components.
+
+- `FingerprintJS.componentsToDebugString(components: object): string`
+
+    Converts a dictionary of components (described above) into human-friendly format.
+
+## Migration from v2
+
+- [Migration guide](https://github.com/fingerprintjs/fingerprintjs/releases/tag/v3.0.0)
```

Comment:
I'll put the a migration guide to the releases section of the GitHub repository.

Also I suggest to change the version tag format from `1.2.3` to `v1.2.3` because it's an industry standard and the prefix will help to establish automatic deployment in future. @Valve, what do you think about the format change?

Reply:
ok, let's use v1.2.3, thanks